src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodToStateMachineRewriter.cs

```diff
@@ -116,6 +116,22 @@ internal void GenerateMoveNext(BoundStatement body, MethodSymbol moveNextMethod)
         {
             F.CurrentMethod = moveNextMethod;
 
+            BoundStatement newBody;
+            try
+            {
+                newBody = GenerateMoveNextBody(body);
+            }
+            catch (SyntheticBoundNodeFactory.MissingPredefinedMember ex)
```

Comment:
>catch (SyntheticBoundNodeFactory.MissingPredefinedMember ex) [](start = 12, length = 60)

Would it be better to add the try/catch around the entire async rewrite instead? That way we would be able to catch other potentially unprotected code paths.